src/main/ts/utils/Git.ts

```diff
@@ -117,3 +117,21 @@ const detectGitUrlFromDir = async (dir: string): Promise<string> => {
 
 export const resolveGitUrl = async (gitUrlArg: Option<string>, workingDirArg: string): Promise<string> =>
   O.isSome(gitUrlArg) ? gitUrlArg.value : await detectGitUrlFromDir(workingDirArg);
+
+const isWorkingDirDirty = async (git: SimpleGit) => {
+  const diff = await git.diffSummary([ 'HEAD' ]);
+  return diff.changed > 0;
+};
+
+const hasUnPushedChanges = async (git: SimpleGit, branchName: string) => {
```

Comment:
Should the `await git.fetch();` also be in this function? as without that this won't work properly.